pkg/operator/ceph/object/objectstore.go

```diff
@@ -289,6 +290,9 @@ func createSimilarPools(context *Context, pools []string, poolSpec model.Pool) e
 			if err != nil {
 				return errors.Errorf("failed to create pool %s for object store %s", name, context.Name)
 			}
+		} else {
+			//update the pool if it already exists
+			ceph.SetPoolProperty(context.Context, context.ClusterName, name, "size", strconv.Itoa(int(poolDetails.Size)))
```

Comment:
Are you always setting this size for existing pools? If it's an erasure-coded pool we wouldn't want to set the size.

Reply:
And why are we setting the pool size to the same thing as `poolDetails.Size`? Seems like no setting is changed. If the size of the pool in the CR changed, we need to get the new size from the CR.

Reply:
> Are you always setting this size for existing pools? If it's an erasure-coded pool we wouldn't want to set the size.

Yes, we are always checking the size.
I forgot about if its an erasure-coded pool or not.
I will add a check accordingly.

Reply:
> And why are we setting the pool size to the same thing as `poolDetails.Size`? Seems like no setting is changed. If the size of the pool in the CR changed, we need to get the new size from the CR.

`setPoolProperty` method is used to change the pool's property right?.  If I change the pool property here i.e the pool's size, that will be updated in CR  too I believe.

Reply:
> setPoolProperty method is used to change the pool's property right?. If I change the pool property here i.e the pool's size, that will be updated in CR too I believe.

The CR is only updated by the admin when they want to change the desired state. The operator then responds to the action of updating the settings in the CR by running this orchestration. We want to make sure the settings are applied from the CR. In this case, I would expect nothing to happen if you're applying the setting to Ceph that you just retrieved from Ceph.

Reply:
To be more specific, I'd recommend the following in this else block:
```
} else {
  logger.Infof("pool %s already exists", name)
  if !ecPool {
      replicationChanged := //detect if the replication is different from the pool details
      if replicationChanged {
        // call SetPoolReplicatedSizeProperty() similar to how it is called in the CreateReplicatedPoolForApp() method
      } 
  } 
}
```